I don't know what to title this.

I noticed players shooting each other in a warzone with flame bows. This error message spammed my console. I can't figure out what is wrong.
[21:57:47] [Server thread/ERROR]: Could not pass event EntityCombustByEntityEvent to Factions v1.6.9.5-U0.1.19-RC2org.bukkit.event.EventExceptionat org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigotmc-1.11-R0.1.jar:git-Spigot-f950f8e-4d99fed]at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigotmc-1.11-R0.1.jar:git-Spigot-f950f8e-4d99fed]at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigotmc-1.11-R0.1.jar:git-Spigot-f950f8e-4d99fed]at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigotmc-1.11-R0.1.jar:git-Spigot-f950f8e-4d99fed]at net.minecraft.server.v1_11_R1.EntityArrow.a(EntityArrow.java:279) [spigotmc-1.11-R0.1.jar:git-Spigot-f950f8e-4d99fed]at net.minecraft.server.v1_11_R1.EntityArrow.A_(EntityArrow.java:194) [spigotmc-1.11-R0.1.jar:git-Spigot-f950f8e-4d99fed]at net.minecraft.server.v1_11_R1.EntityTippedArrow.A_(EntityTippedArrow.java:72) [spigotmc-1.11-R0.1.jar:git-Spigot-f950f8e-4d99fed]at net.minecraft.server.v1_11_R1.World.entityJoinedWorld(World.java:1652) [spigotmc-1.11-R0.1.jar:git-Spigot-f950f8e-4d99fed]at net.minecraft.server.v1_11_R1.World.g(World.java:1627) [spigotmc-1.11-R0.1.jar:git-Spigot-f950f8e-4d99fed]at net.minecraft.server.v1_11_R1.World.tickEntities(World.java:1453) [spigotmc-1.11-R0.1.jar:git-Spigot-f950f8e-4d99fed]at net.minecraft.server.v1_11_R1.WorldServer.tickEntities(WorldServer.java:635) [spigotmc-1.11-R0.1.jar:git-Spigot-f950f8e-4d99fed]at net.minecraft.server.v1_11_R1.MinecraftServer.D(MinecraftServer.java:806) [spigotmc-1.11-R0.1.jar:git-Spigot-f950f8e-4d99fed]at net.minecraft.server.v1_11_R1.DedicatedServer.D(DedicatedServer.java:399) [spigotmc-1.11-R0.1.jar:git-Spigot-f950f8e-4d99fed]at net.minecraft.server.v1_11_R1.MinecraftServer.C(MinecraftServer.java:675) [spigotmc-1.11-R0.1.jar:git-Spigot-f950f8e-4d99fed]at net.minecraft.server.v1_11_R1.MinecraftServer.run(MinecraftServer.java:574) [spigotmc-1.11-R0.1.jar:git-Spigot-f950f8e-4d99fed]at java.lang.Thread.run(Thread.java:745) [?:1.8.0_112]Caused by: java.lang.NoSuchMethodError: org.bukkit.event.entity.EntityDamageByEntityEvent.(Lorg/bukkit/entity/Entity;Lorg/bukkit/entity/Entity;Lorg/bukkit/event/entity/EntityDamageEvent$DamageCause;I)Vat com.massivecraft.factions.listeners.FactionsEntityListener.onEntityCombustByEntity(FactionsEntityListener.java:201) ~[?:?]at sun.reflect.GeneratedMethodAccessor535.invoke(Unknown Source) ~[?:?]at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_112]at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_112]at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[spigotmc-1.11-R0.1.jar:git-Spigot-f950f8e-4d99fed]... 15 more

--- a/pom.xml
+++ b/pom.xml
@@ -73,7 +73,7 @@
         <dependency>
             <groupId>org.spigotmc</groupId>
             <artifactId>spigot-api</artifactId>
-            <version>1.9.4-R0.1-SNAPSHOT</version>
+            <version>1.11-R0.1-SNAPSHOT</version>
             <scope>provided</scope>
             <exclusions>
             	<exclusion>

--- a/src/main/java/com/massivecraft/factions/cmd/CmdList.java
+++ b/src/main/java/com/massivecraft/factions/cmd/CmdList.java
@@ -3,13 +3,12 @@ package com.massivecraft.factions.cmd;
 import com.massivecraft.factions.Conf;
 import com.massivecraft.factions.Faction;
 import com.massivecraft.factions.Factions;
+import com.massivecraft.factions.P;
 import com.massivecraft.factions.struct.Permission;
 import com.massivecraft.factions.zcore.util.TL;
 import com.massivecraft.factions.zcore.util.TagUtil;
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.Comparator;
+import java.util.*;
 
 
 public class CmdList extends FCommand {
@@ -49,6 +48,18 @@ public class CmdList extends FCommand {
         factionList.remove(Factions.getInstance().getNone());
         factionList.remove(Factions.getInstance().getSafeZone());
         factionList.remove(Factions.getInstance().getWarZone());
+
+        // remove exempt factions
+        if (!fme.getPlayer().hasPermission("factions.show.bypassexempt")) {
+            List<String> exemptFactions = P.p.getConfig().getStringList("show-exempt");
+            Iterator<Faction> factionIterator = factionList.iterator();
+            while (factionIterator.hasNext()) {
+                Faction next = factionIterator.next();
+                if (exemptFactions.contains(next.getTag())) {
+                    factionIterator.remove();
+                }
+            }
+        }
 
         // Sort by total followers first
         Collections.sort(factionList, new Comparator<Faction>() {

--- a/src/main/java/com/massivecraft/factions/cmd/CmdShow.java
+++ b/src/main/java/com/massivecraft/factions/cmd/CmdShow.java
@@ -54,6 +54,12 @@ public class CmdShow extends FCommand {
             return;
         }
 
+        if (!fme.getPlayer().hasPermission("factions.show.bypassexempt")
+                && P.p.getConfig().getStringList("show-exempt").contains(faction.getTag())) {
+            msg(TL.COMMAND_SHOW_EXEMPT);
+            return;
+        }
+
         // if economy is enabled, they're not on the bypass list, and this command has a cost set, make 'em pay
         if (!payForCommand(Conf.econCostShow, TL.COMMAND_SHOW_TOSHOW, TL.COMMAND_SHOW_FORSHOW)) {
             return;

--- a/src/main/java/com/massivecraft/factions/listeners/FactionsEntityListener.java
+++ b/src/main/java/com/massivecraft/factions/listeners/FactionsEntityListener.java
@@ -198,7 +198,7 @@ public class FactionsEntityListener implements Listener {
     // mainly for flaming arrows; don't want allies or people in safe zones to be ignited even after damage event is cancelled
     @EventHandler(priority = EventPriority.NORMAL, ignoreCancelled = true)
     public void onEntityCombustByEntity(EntityCombustByEntityEvent event) {
-        EntityDamageByEntityEvent sub = new EntityDamageByEntityEvent(event.getCombuster(), event.getEntity(), EntityDamageEvent.DamageCause.FIRE, 0);
+        EntityDamageByEntityEvent sub = new EntityDamageByEntityEvent(event.getCombuster(), event.getEntity(), EntityDamageEvent.DamageCause.FIRE, 0d);
         if (!this.canDamagerHurtDamagee(sub, false)) {
             event.setCancelled(true);
         }

--- a/src/main/java/com/massivecraft/factions/zcore/util/TL.java
+++ b/src/main/java/com/massivecraft/factions/zcore/util/TL.java
@@ -463,6 +463,7 @@ public enum TL {
     COMMAND_SHOW_MEMBERSOFFLINE("Members offline: "),
     COMMAND_SHOW_COMMANDDESCRIPTION("Show faction information"),
     COMMAND_SHOW_DEATHS_TIL_RAIDABLE("<i>DTR: %1$d"),
+    COMMAND_SHOW_EXEMPT("<b>This faction is exempt and cannot be seen."),
 
     COMMAND_SHOWINVITES_PENDING("Players with pending invites: "),
     COMMAND_SHOWINVITES_CLICKTOREVOKE("Click to revoke invite for %1$s"),

--- a/src/main/resources/config.yml
+++ b/src/main/resources/config.yml
@@ -245,6 +245,11 @@ show:
 # For a /f show that does not display fancy messages that are essentially empty, use minimal-show
 minimal-show: false
 
+# Factions that should be exempt from /f show, case sensitive, useful for a
+# serverteam faction, since the command shows vanished players otherwise
+show-exempt:
+  - Put_faction_tag_here
+
 ############################################################
 # +------------------------------------------------------+ #
 # |                Configurable /f list                  | #

--- a/src/main/resources/plugin.yml
+++ b/src/main/resources/plugin.yml
@@ -48,6 +48,7 @@ permissions:
       factions.kick.any: true
       factions.ownershipbypass: true
       factions.chatspy: true
+      factions.show.bypassexempt: true
   factions.kit.fullplayer:
     default: true
     description: Can also create new factions.
@@ -231,6 +232,8 @@ permissions:
     description: set faction home for another faction
   factions.show:
     description: show faction information
+  factions.show.bypassexempt:
+    description: show exempt faction information
   factions.tag:
     description: change the faction tag
   factions.title:

